apis/compute/v1beta1/gkecluster_types.go

```diff
@@ -0,0 +1,1686 @@
+	// `container_ipv4_cidr`
+	// range. This field will only be set when cluster is in route-based
+	// network
+	// mode.
+	NodeIpv4CidrSize int64 `json:"nodeIpv4CidrSize,omitempty"`
+
+	// PrivateClusterConfig: Configuration for private cluster.
+	PrivateClusterConfig *PrivateClusterConfigStatus `json:"privateClusterConfig,omitempty"`
+
+	// NOTE(hasheddan): node pools are modelled in status only because
+	// management of node pools is handled by the stack-gcp NodePool object.
+
+	// NodePools: The node pools associated with this cluster.
+	// This field should not be set if "node_config" or "initial_node_count"
+	// are
+	// specified.
+	NodePools []*NodePoolClusterStatus `json:"nodePools,omitempty"`
+
+	// SelfLink: Server-defined URL for the resource.
+	SelfLink string `json:"selfLink,omitempty"`
+
+	// ServicesIpv4Cidr: The IP address range of the
+	// Kubernetes services in
+	// this cluster,
+	// in
+	// [CIDR](http://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing)
+	//
+	// notation (e.g. `1.2.3.4/29`). Service addresses are
+	// typically put in the last `/16` from the container CIDR.
+	ServicesIpv4Cidr string `json:"servicesIpv4Cidr,omitempty"`
+
+	// Status: The current status of this cluster.
+	//
+	// Possible values:
+	//   "STATUS_UNSPECIFIED" - Not set.
+	//   "PROVISIONING" - The PROVISIONING state indicates the cluster is
+	// being created.
+	//   "RUNNING" - The RUNNING state indicates the cluster has been
+	// created and is fully
+	// usable.
+	//   "RECONCILING" - The RECONCILING state indicates that some work is
+	// actively being done on
+	// the cluster, such as upgrading the master or node software. Details
+	// can
+	// be found in the `statusMessage` field.
+	//   "STOPPING" - The STOPPING state indicates the cluster is being
+	// deleted.
+	//   "ERROR" - The ERROR state indicates the cluster may be unusable.
+	// Details
+	// can be found in the `statusMessage` field.
+	//   "DEGRADED" - The DEGRADED state indicates the cluster requires user
+	// action to restore
+	// full functionality. Details can be found in the `statusMessage`
+	// field.
+	Status string `json:"status,omitempty"`
+
+	// StatusMessage: Additional information about the current
+	// status of this
+	// cluster, if available.
+	StatusMessage string `json:"statusMessage,omitempty"`
+
+	// TpuIpv4CidrBlock: The IP address range of the Cloud
+	// TPUs in this cluster,
+	// in
+	// [CIDR](http://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing)
+	//
+	// notation (e.g. `1.2.3.4/29`).
+	TpuIpv4CidrBlock string `json:"tpuIpv4CidrBlock,omitempty"`
+
+	// Zone: The name of the Google Compute
+	// Engine
+	// [zone](/compute/docs/zones#available) in which the
+	// cluster
+	// resides.
+	// This field is deprecated, use location instead.
+	Zone string `json:"zone,omitempty"`
+}
+
+// GKEClusterParameters define the desired state of a Google Kubernetes Engine
+// cluster.
+type GKEClusterParameters struct {
```

Comment:
Not a strong opinion: I think the required fields could be at the top so that our doc generation tooling can put those fields in the top. Otherwise, it's a little bit hard to identify which fields are required.